Source: upstream
Upstream: yes
Reason: Check correctly before including SSE4.2 header.

commit 9f18a1ad5ce32dd397642a4c03fa1fcb21fb9456
Author: Mike McQuaid <mike@mikemcquaid.com>
Date:   Thu Nov 18 05:40:36 2010 +0000

    Check correctly before including SSE4.2 header.
    
    Previously, this failed machines that only had SSE4.1 but not
    SSE4.2 due to an assumption that nmmintrin.h could be included
    when only using SSE4.1.
    
    Fixes http://bugreports.qt.nokia.com/browse/QTBUG-13623.
    
    Merge-request: 929
    Reviewed-by: Benjamin Poulain <benjamin.poulain@nokia.com>

diff --git a/src/corelib/tools/qsimd_p.h b/src/corelib/tools/qsimd_p.h
index 87fa770..2dbed76 100644
--- a/src/corelib/tools/qsimd_p.h
+++ b/src/corelib/tools/qsimd_p.h
@@ -87,9 +87,13 @@ QT_BEGIN_HEADER
 #include <tmmintrin.h>
 #endif
 
-// SSE4.1 and SSE4.2 intrinsics
-#if (defined(QT_HAVE_SSE4_1) || defined(QT_HAVE_SSE4_2)) && (defined(__SSE4_1__) || defined(Q_CC_MSVC))
+// SSE4.1 intrinsics
+#if defined(QT_HAVE_SSE4_1) && (defined(__SSE4_1__) || defined(Q_CC_MSVC))
 #include <smmintrin.h>
+#endif
+
+// SSE4.2 intrinsics
+#if defined(QT_HAVE_SSE4_2) && (defined(__SSE4_2__) || defined(Q_CC_MSVC))
 #include <nmmintrin.h>
 #endif
 
